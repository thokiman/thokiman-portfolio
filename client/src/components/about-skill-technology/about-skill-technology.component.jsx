import React from "react";
import { CSSTransitionGroup } from "react-transition-group";

import TechnologyGrid from "../technology-grid/technology-grid.component";
import {
  AboutTechnologyContentContainer,
  AboutTechnologySkillContainer,
} from "./about-skill-technology.styles";

const TechnologySkill = ({ skill }) => {
  const {
    itemsPoint: {
      general,
      content,
      frontend: { html, css, javascript },
      backend: { language, framework, database },
      architecture,
      test,
    },
  } = skill;
  return (
    <AboutTechnologySkillContainer>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={200}
        transitionLeaveTimeout={200}
        transitionName={"slide-out"}
      >
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={general} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={content} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={html} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={css} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={javascript} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={language} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={framework} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={database} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={architecture} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={test} />
        </AboutTechnologyContentContainer>
      </CSSTransitionGroup>
    </AboutTechnologySkillContainer>
  );
};

export default TechnologySkill;
