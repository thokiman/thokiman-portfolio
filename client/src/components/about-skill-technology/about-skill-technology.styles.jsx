import styled from "styled-components";

export const AboutTechnologySkillContainer = styled.div`
  width: 95%;
  height: 88%;
  overflow-y: scroll;
  overflow-x: hidden;
  margin: 0 auto;
  @media only screen and (max-width: 800px) {
    width: 100%;
  }
`;
export const AboutTechnologyContentContainer = styled.div`
  margin: 20px auto;
  border-top: #74c6d4 solid 2px;
  border-bottom: #74c6d4 solid 2px;
`;

AboutTechnologySkillContainer.displayName = "AboutTechnologySkillContainer";
AboutTechnologyContentContainer.displayName = "AboutTechnologyContentContainer";
