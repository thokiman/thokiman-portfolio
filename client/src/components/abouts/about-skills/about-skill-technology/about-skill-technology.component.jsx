import React from 'react';
import { CSSTransitionGroup } from 'react-transition-group';
import TechnologyGrid from '../../about-commons/skill-parts/technology-skill-parts/technology-grid/technology-grid.component';
import {
  AboutTechnologyContentContainer,
  AboutTechnologySkillContainer
} from './about-skill-technology.styles';

export const TechnologySkill = ({ skill }) => {
  console.log(skill);
  const {
    itemsPoint: {
      architecture,
      backend: { database, framework },
      content,
      deploy,
      desktop,
      frontend: { html, css, javascript },
      general,
      language,
      library,
      test,
      webscraping,
    },
  } = skill;
  return (
    <AboutTechnologySkillContainer>
      <CSSTransitionGroup
        transitionAppear={true}
        transitionAppearTimeout={800}
        transitionEnterTimeout={200}
        transitionLeaveTimeout={200}
        transitionName={'slide-out'}
      >
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={architecture} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={framework} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={database} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={content} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={deploy} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={desktop} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={html} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={css} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={javascript} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={general} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={language} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={library} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={test} />
        </AboutTechnologyContentContainer>
        <AboutTechnologyContentContainer>
          <TechnologyGrid technologySkill={webscraping} />
        </AboutTechnologyContentContainer>
      </CSSTransitionGroup>
    </AboutTechnologySkillContainer>
  );
};

export default TechnologySkill;
