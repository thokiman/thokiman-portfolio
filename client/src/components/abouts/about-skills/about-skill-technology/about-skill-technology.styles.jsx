import styled from 'styled-components';

const maxWidthSmartphone = 800;
export const AboutTechnologySkillContainer = styled.div`
  width: 95%;
  height: 88%;
  overflow-y: scroll;
  overflow-x: hidden;
  margin: 0 auto;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 100%;
    overflow-y: hidden;
  }
`;
export const AboutTechnologyContentContainer = styled.div`
  margin: 20px auto;
  padding: 20px 0px;
`;

AboutTechnologySkillContainer.displayName = 'AboutTechnologySkillContainer';
AboutTechnologyContentContainer.displayName = 'AboutTechnologyContentContainer';
