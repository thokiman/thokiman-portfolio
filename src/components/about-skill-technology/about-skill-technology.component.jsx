import React from "react";

import TechnologyGrid from "../technology-grid/technology-grid.component";
import "./about-skill-technology.styles.scss";

const TechnologySkill = ({ skill }) => {
  const {
    itemsPoint: {
      general,
      content,
      frontend: { html, css, javascript },
      backend: { language, framework, database },
      architecture,
      test,
    },
  } = skill;
  return (
    <div className="technology-skill-container">
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={general} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={content} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={html} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={css} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={javascript} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={language} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={framework} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={database} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={architecture} />
      </div>
      <div className="technology-content-container">
        <TechnologyGrid technologySkill={test} />
      </div>
    </div>
  );
};

export default TechnologySkill;
