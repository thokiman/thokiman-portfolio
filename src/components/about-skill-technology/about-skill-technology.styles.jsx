import styled from "styled-components";

export const AboutTechnologySkillContainer = styled.div`
  width: 95%;
  height: 88%;
  overflow-y: scroll;
  overflow-x: hidden;
  margin: 0 auto;
`;
export const AboutTechnologyContentContainer = styled.div`
  position: relative;
  top: 5px;
  margin: 10px auto;
  border-top: #74c6d4 solid 2px;
  border-bottom: #74c6d4 solid 2px;
`;

AboutTechnologySkillContainer.displayName = "AboutTechnologySkillContainer";
AboutTechnologyContentContainer.displayName = "AboutTechnologyContentContainer";
