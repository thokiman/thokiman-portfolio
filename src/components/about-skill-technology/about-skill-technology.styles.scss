.technology-skill-container {

    width: 95%;
    height: 88%;
    overflow-y: scroll;
    overflow-x: hidden;
    margin: auto 20px;

    .technology-content-container {
        position: relative;
        top: 5px;
        margin: 10px auto;
        border-top: #74c6d4 solid 2px;
        border-bottom: #74c6d4 solid 2px;
    }


}