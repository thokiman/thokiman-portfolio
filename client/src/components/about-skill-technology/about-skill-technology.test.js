import React from "react";
import { shallow } from "enzyme";

import { TechnologySkill } from "./about-skill-technology.component";

describe("TechnologySkill component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      skill: {
        itemsPoint: {
          general: mockArray,
          content: mockArray,
          frontend: { html: mockArray, css: mockArray, javascript: mockArray },
          backend: {
            language: mockArray,
            framework: mockArray,
            database: mockArray,
          },
          architecture: mockArray,
          test: mockArray,
        },
      },
    };
    wrapper = shallow(<TechnologySkill {...mockProps} />);
  });

  it("should render TechnologySkill component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
