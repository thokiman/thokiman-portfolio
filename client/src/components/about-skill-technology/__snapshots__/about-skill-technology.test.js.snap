// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologySkill component should render TechnologySkill component 1`] = `ShallowWrapper {}`;
